Dockerfile

```
@@ -0,0 +1,54 @@
+    CGO_ENABLED=0 go build -ldflags='-s -w' -o /out/tracera-backend .
+
+FROM node:${NODE_VERSION}-alpine AS frontend-builder
+WORKDIR /build
+COPY frontend/package.json frontend/bun.lockb* ./
+RUN corepack enable && bun install --frozen-lockfile || npm ci
+COPY frontend/ ./
+RUN bun run build || npm run build
+
+FROM python:${PYTHON_VERSION}-slim AS final
+WORKDIR /app
+RUN pip install --no-cache-dir uv
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)